health-services/individual/src/main/java/org/egov/individual/service/NonExistentEntityValidator.java

```diff
@@ -0,0 +1,63 @@
+package org.egov.individual.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.egov.common.models.Error;
+import org.egov.common.utils.Validator;
+import org.egov.individual.repository.IndividualRepository;
+import org.egov.individual.web.models.Individual;
+import org.egov.individual.web.models.IndividualBulkRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.egov.common.utils.CommonUtils.checkNonExistentEntities;
+import static org.egov.common.utils.CommonUtils.getIdFieldName;
+import static org.egov.common.utils.CommonUtils.getIdToObjMap;
+import static org.egov.common.utils.CommonUtils.getMethod;
+import static org.egov.common.utils.CommonUtils.getObjClass;
+import static org.egov.common.utils.CommonUtils.notHavingErrors;
+import static org.egov.common.utils.CommonUtils.populateErrorDetails;
+import static org.egov.common.utils.ValidatorUtils.getErrorForNonExistentEntity;
+
+@Component
+@Order(value = 4)
+@Slf4j
+public class NonExistentEntityValidator implements Validator<IndividualBulkRequest, Individual> {
+
+    private final IndividualRepository individualRepository;
+
+    @Autowired
+    public NonExistentEntityValidator(IndividualRepository individualRepository) {
+        this.individualRepository = individualRepository;
+    }
+
+
+    @Override
```

Comment:
Not required for sub-entities as we are relying on postgres on conflict update clause.

Reply:
But then client won't know what to correct in data as the data will not be in error queue

Reply:
Client doesn't need to know that. If an address doesn't exist, it will be created. If an address exists, it will be updated. Technical validations are being performed anyways by spring boot. I hope that clears it.